src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/SqlManagerController.php

```diff
@@ -0,0 +1,337 @@
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2017 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Controller\Admin\Configure\AdvancedParameters;
+
+use PrestaShop\PrestaShop\Core\Form\FormHandlerInterface;
+use PrestaShop\PrestaShop\Core\Table\Column;
+use PrestaShop\PrestaShop\Core\Table\Factory\TableViewFactory;
+use PrestaShop\PrestaShop\Core\Table\RowAction;
+use PrestaShop\PrestaShop\Core\Table\Table;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Form\Admin\Configure\AdvancedParameters\SqlManager\FilterRequestSqlType;
+use PrestaShopBundle\Security\Voter\PageVoter;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Responsible of "Configure > Advanced Parameters > Database -> SQL Manager" page
+ */
+class SqlManagerController extends FrameworkBundleAdminController
+{
+    /**
+     * Show list of saved SQL's
+     *
+     * @Template("@PrestaShop/Admin/Configure/AdvancedParameters/SqlManager/list.html.twig")
+     *
+     * @param Request $request
+     *
+     * @return array
+     */
+    public function listAction(Request $request)
+    {
+        $searchForm = $this->createForm(FilterRequestSqlType::class, []);
+        $searchForm->handleRequest($request);
+
+        $filters = $this->get('prestashop.core.admin.search_parameters')->getFiltersFromRequest($request, [
+            'limit' => 10,
+            'offset' => 0,
+            'orderBy' => 'id_request_sql',
+            'sortOrder' => 'desc',
+            'filters' => $searchForm->getData(),
+        ]);
+
+        $repository = $this->getRepository();
+        $requestSqls = $repository->findByFilters($filters);
+        $requestSqlsCount = $repository->getCount();
+
+        $settingsForm = $this->getSettingsFormHandler()->getForm();
+
+        $data = [
+            'request_sqls' => $requestSqls,
+            'request_sqls_count' => $requestSqlsCount,
+            'order_by' => $filters['orderBy'],
+            'order_way' => $filters['sortOrder'],
+            'filters' => $filters,
+            'search_form' => $searchForm->createView(),
+            'settings_form' => $settingsForm->createView(),
+        ];
+
+        return $this->getTemplateParams($request) + $data;
+    }
+
+    /**
+     * @Template("@PrestaShop/Admin/Configure/AdvancedParameters/SqlManager/table.html.twig")
```

Comment:
don't you think we can have only one re-usable template in "Common" for every table ?

Reply:
It has one reusable template under src/PrestaShopBundle/Resources/views/Admin/Common/table.html.twig